tensorrt_llm/_torch/speculative/interface.py

```diff
@@ -85,12 +85,19 @@ def extend_ctx(self, attention_backend: Type[AttentionBackend]):
         chunked context requests at the kernel level. Required for
         any spec dec mode that uses the SpecExecutor.
         """
-
         # Fixme: only trtllm attention backend supports eagle3 generation-phase kernels on blackwell.
-        return ((self.is_eagle3() or self.is_draft_target())
-                and not (isinstance(attention_backend, TrtllmAttention)
-                         and get_sm_version() == 100)
-                ) or self.is_ngram() or self.is_user_provided()
+
+        if self.is_eagle3() or self.is_draft_target():
+            if get_sm_version() == 100 and isinstance(attention_backend,
```

Comment:
Should be `issubclass`. That is technically also applicable to the `attention_backend is TrtllmAttention` below (though the usage of `issubclass` vs `is` is a minor issue, both are functionally the same since subclass of `TrtllmAttention` are not likely)